src/Cli/Exporter.cs

```diff
@@ -31,26 +32,34 @@ public static int Export(ExportOptions options, ILogger logger, FileSystemRuntim
                     out RuntimeConfig? runtimeConfig,
                     replaceEnvVar: true) || runtimeConfig is null)
             {
-                logger.LogError("Failed to read the config file: {runtimeConfigFile}.", runtimeConfigFile);
+                logger.LogError($"Failed to read the config file: {runtimeConfigFile}.");
                 return -1;
             }
 
-            Task server = Task.Run(() =>
+            // Do not retry if schema generation logic is running
+            int retryCount = 1;
+
+            if (!options.Generate)
             {
-                _ = ConfigGenerator.TryStartEngineWithOptions(startOptions, loader, fileSystem);
-            }, cancellationToken);
+                // No need to start graphQL server if schema generation logic is running
+                _ = Task.Run(() =>
+                {
+                    _ = ConfigGenerator.TryStartEngineWithOptions(startOptions, loader, fileSystem);
+                }, cancellationToken);
+
+                retryCount = 5;
+            }
 
             bool isSuccess = false;
             if (options.GraphQL)
             {
-                int retryCount = 5;
                 int tries = 0;
 
                 while (tries < retryCount)
                 {
                     try
                     {
-                        ExportGraphQL(options, runtimeConfig, fileSystem);
+                        ExportGraphQL(options, runtimeConfig, fileSystem, logger).Wait();
```

Comment:
There are only 2 conditions:
1. ExportGraphQL() => it clearly says which logic to run, to generate/fetch the schema
2. Export constructor => it has a condition, where we are starting graphql serve if generation is not requested.

I don't think, it is going to be more complex in future. If it does, we can definitely refactor it. what do you think?

Reply:
To clarify my intent with the comment, you currently have
- `Export(...)` Top Level CLI verb
  - `ExportGraphQL(options, ...)` Helper function which can do both export from CosmosDB AND export from DAB service

It would be cleaner to have separate helper to handle only CosmosDB export:
- `Export(...)` Top Level CLI verb
  - `private ExportGraphQLFromCosmosDB(options, ...)`
  - `private ExportGraphQLFromDabService(options, ...)`